src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -97,78 +133,59 @@ async fn xor_specialized_1<F: Field>(
 ///
 /// And helper 3 has shares:
 /// (x3, 0)
-async fn xor_specialized_2<F: Field>(
-    ctx: SemiHonestContext<'_, F>,
-    record_id: RecordId,
-    a: &Replicated<F>,
-    b: &Replicated<F>,
-) -> Result<Replicated<F>, Error> {
+async fn xor_specialized_2<F, C, S>(ctx: C, record_id: RecordId, a: &S, b: &S) -> Result<S, Error>
+where
+    F: Field,
+    C: Context<F, Share = S>,
+    S: SecretSharing<F>,
+{
     let result = ctx
         .multiply_one_share_mostly_zeroes(record_id, a, b)
         .await?;
 
-    Ok(a + b - &(result * F::from(2)))
+    Ok(-(result * F::from(2)) + a + b)
 }
 
-/// This takes a replicated secret sharing of a sequence of bits (in a packed format)
-/// and converts them, one bit-place at a time, to secret sharings of that bit value (either one or zero) in the target field.
-///
-/// This file is somewhat inspired by Algorithm D.3 from <https://eprint.iacr.org/2018/387.pdf>
-/// "Efficient generation of a pair of random shares for small number of parties"
-///
-/// This protocol takes as input such a 3-way random binary replicated secret-sharing,
-/// and produces a 3-party replicated secret-sharing of the same value in a target field
-/// of the caller's choosing.
-/// Example:
-/// For input binary sharing: (0, 1, 1) -> which is a sharing of 0 in `Z_2`
-/// sample output in `Z_31` could be: (22, 19, 21) -> also a sharing of 0 in `Z_31`
-/// This transformation is simple:
-/// The original can be conceived of as r = b0 ⊕ b1 ⊕ b2
-/// Each of the 3 bits can be trivially converted into a 3-way secret sharing in `Z_p`
-/// So if the second bit is a '1', we can make a 3-way secret sharing of '1' in `Z_p`
-/// as (0, 1, 0).
-/// Now we simply need to XOR these three sharings together in `Z_p`. This is easy because
-/// we know the secret-shared values are all either 0, or 1. As such, the XOR operation
-/// is equivalent to fn xor(a, b) { a + b - 2*a*b }
-pub async fn convert_one_bit<F: Field>(
-    ctx: SemiHonestContext<'_, F>,
-    record_id: RecordId,
-    input: &XorReplicated,
-    bit_index: u32,
-) -> Result<Replicated<F>, Error> {
-    let [sh0, sh1, sh2] = local_secret_share(&input.bit(bit_index), ctx.role());
-
-    let sh0_xor_sh1 = xor_specialized_1(ctx.narrow(&Step::Xor1), record_id, &sh0, &sh1).await?;
-    xor_specialized_2(ctx.narrow(&Step::Xor2), record_id, &sh0_xor_sh1, &sh2).await
+pub async fn convert_bit<F, C, S>(ctx: C, record_id: RecordId, input: &[S; 3]) -> Result<S, Error>
```

Comment:
What do you think of: `convert_bit_modulus`?

Reply:
I think that I don't like the name `input`. How about something like `locally_converted_bit_shares` or `locally_generated_bit_shares`